src/plugins/timelion/server/plugin.ts

```diff
@@ -17,44 +17,50 @@
  * under the License.
  */
 
-import { CoreSetup, Plugin, PluginInitializerContext } from 'src/core/server';
+import { CoreSetup, CoreStart, Plugin, PluginInitializerContext, Logger } from 'src/core/server';
 import { i18n } from '@kbn/i18n';
 import { schema } from '@kbn/config-schema';
 import { TimelionConfigType } from './config';
+import { timelionSheetSavedObjectType } from './saved_objects';
+
+/**
+ * Timelion plugin was deprecated since 7.0, the Timelion app will be removed in 8.0.
+ * To continue using your Timelion worksheets, migrate them to a dashboard.
+ *
+ *  @link https://www.elastic.co/guide/en/kibana/master/timelion.html#timelion-deprecation
+ **/
+const showWarningMessageIfTimelionSteetWasFound = (core: CoreStart, logger: Logger) => {
+  const { savedObjects } = core;
+  const savedObjectsClient = savedObjects.createInternalRepository();
+
+  savedObjectsClient
+    .find({
+      type: 'timelion-sheet',
+      perPage: 1,
+    })
+    .then(
+      ({ total }) =>
+        total &&
+        logger.warn(
+          'Timelion plugin was deprecated since 7.0, the Timelion app will be removed in 8.0.'
```

Comment:
Let's be more verbose here, that they still have timelion worksheets saved and maybe directly output the link to the docuemntation.

I'd suggest the following (cc @gchaps and @KOTungseth for final message):

```
The timelion application was deprecated since 7.0 and will be removed in 8.0. You still have saved timelion application worksheets, which you need to migrate manually to dashboards if you want to keep them after 8.0. See https://www.elastic.co/guide/en/kibana/master/timelion.html#timelion-deprecation for more details.
```

It seems that the doc_links service is not available on the serverside to generate links. That would mean we need to hardcode that link here, unless Platform team has still plans to expose it also server side (cc @joshdover could you give some details on this?)

Reply:
The warning we have on the app could be used as was suggested by @gchaps 
`Deprecated since 7.0, the Timelion app will be removed in 8.0. To continue using your Timelion worksheets, migrate them to a dashboard. See https://www.elastic.co/guide/en/kibana/master/timelion.html#timelion-deprecation `

Reply:
@stratoula @timroes it was initially like on app. Then I updated it as Tim suggested. 
Does it mean that I should return it back? Actually I like 2 options. Probably we can keep it as it

Reply:
No don't return it back! Let's wait for @gchaps and @KOTungseth for the final message

Reply:
@timroes, @alexwizp 

I agree with @stratoula that the text in the message and docs should be the same.  If this text suits your needs, then I would go with it: 

>Deprecated since 7.0, the Timelion app will be removed in 8.0. To continue using your Timelion worksheets, migrate them to a dashboard. See https://www.elastic.co/guide/en/kibana/master/timelion.html#timelion-deprecation.